include/nbl/video/ISwapchain.h

```diff
@@ -99,11 +103,18 @@ class NBL_API ISwapchain : public core::IReferenceCounted, public IBackendObject
         // Vulkan: const VkSwapchainKHR*
         virtual const void* getNativeHandle() const = 0;
 
+        void freeImageExists(uint32_t ix) { m_imageExists->begin()[ix].store(false); }
+
     protected:
         virtual ~ISwapchain() = default;
 
         SCreationParams m_params;
         images_array_t m_images;
+        core::smart_refctd_dynamic_array<std::atomic<bool>> m_imageExists;
```

Comment:
eh, use one `atomic_uint32_t`, then use 1 bit per image (ergo max 32 images for any swapchain)